src/Cms/Plugin.php

```diff
@@ -40,16 +34,44 @@ class Plugin
 	 *
 	 * @throws \Kirby\Exception\InvalidArgumentException If the plugin name has an invalid format
 	 */
-	public function __construct(string $name, array $extends = [])
-	{
+	public function __construct(
+		protected string $name,
+		protected array $extends = [],
+		protected array $info = [],
+		protected string|null $root = null,
+		protected string|null $version = null,
+	) {
 		static::validateName($name);
 
-		$this->name    = $name;
-		$this->extends = $extends;
-		$this->root    = $extends['root'] ?? dirname(debug_backtrace()[0]['file']);
-		$this->info    = empty($extends['info']) === false && is_array($extends['info']) ? $extends['info'] : null;
+		// TODO: Remove in v5
+		if ($root = $extends['root'] ?? null) {
+			Helpers::deprecated('Plugin "' . $name . '": Passing the `root` inside the `extends` array has been deprecated. Pass it directly as named argument `root`.', 'plugin-extends-root');
+			$this->root ??= $root;
+			unset($this->extends['root']);
+		}
+
+		$this->root ??= dirname(debug_backtrace()[0]['file']);
+
+		// TODO: Remove info property in v5
```

Comment:
I think you meant the fallback, not the property:

```suggestion
		// TODO: Remove in v5
```